yamux/src/connection/stream.rs

```diff
@@ -360,15 +541,21 @@ impl AsyncWrite for Stream {
                 log::debug!("{}/{}: can no longer write", self.conn, self.id);
                 return Poll::Ready(Err(self.write_zero_err()));
             }
-            if shared.credit == 0 {
+            if shared.current_send_window_size == 0 {
```

Comment:
But neither `window` nor `credit` implies the direction. The `window` on one side is the `credit` of the other. Which one is which? The names don't imply that from the local perspective is inbound and `credit` is outbound.

Reply:
I don't really mind what we are naming them but I find 4 words too long for this variable, esp. if any related state has 3 out of 4 in common. Just conceptually, this means those 3 words don't add value in differentiating the two from each other.

We never have an "old" or "outdated" window size so `current` is meaningless IMO. `size` doesn't add much either because with in the context of yamux, `send_window` and `receive_window` are expressions of quantities so it is implicit that we are measuring their size.

So maybe just `send_window` and `receive_window`?